sdk/core/core-client-rest/src/multipart.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  BodyPart,
+  MultipartRequestBody,
+  RestError,
+  createHttpHeaders,
+} from "@azure/core-rest-pipeline";
+import { stringToUint8Array } from "@azure/core-util";
+import { isBinaryBody } from "./helpers/isBinaryBody.js";
+
+/**
+ * Describes a single part in a multipart body.
+ */
+export interface PartDescriptor {
+  /**
+   * Content type of this part. If set, this value will be used to set the Content-Type MIME header for this part, although explicitly
+   * setting the Content-Type header in the headers bag will override this value. If set to `null`, no content type will be inferred from
+   * the body field. Otherwise, the value of the Content-Type MIME header will be inferred based on the type of the body.
+   */
+  contentType?: string | null;
```

Comment:
what if the null is conflicted with headers?

Reply:
If the header is present, the header value is used, regardless of the value of the `contentType` field. That includes `null`.